### PR TITLE
Add support of WebDAV's HTTP Methods for Distributed Authoring

### LINKED ISSUE #937: Support WebDAV's HTTP Methods for Distributed Authoring
It would be great to support WebDAV's HTTP methods for distributed authoring as specified in the [RFC2518 Section 8](https://www.ietf.org/rfc/rfc2518.html#section-8).

The following methods are missing to get support WebDAV:
- `PROPFIND`
- `PROPPATCH`
- `MKCOL`
- `COPY`
- `MOVE`
- `LOCK`
- `UNLOCK`

Currently, the only way to mock a WebDAV endpoints is to use the `Logs` tab of Mockoon and add it to the requests, or by changing the `JSON` manually. However, due to the schema validation the changes in the environment file will be reverted after closing Mockoon.

### PR DESCRIPTION
**Pull request by daniel-schulz**
Added the following methods:
* PROPFIND
* PROPPATCH
* MKCOL
* COPY
* MOVE
* LOCK
* UNLOCK

Closes #937

**Technical implementation details**

**Checklist**

- [x] data migration added (@mockoon/commons) `=> No migration relevant schema changes.`
- [x] data migration automated tests added (@mockoon/commons) `=> No migration relevant schema changes.`
- [x] CLI automated tests added (@mockoon/cli) `=> No changes to the CLI.`
- [x] desktop automated tests added (@mockoon/desktop) `=> As discussed in this pr, no additional tests necessary.`

### CODE REVIEW

File: packages/desktop/src/renderer/styles/styles.scss [hunk 1]
```diff
@@ -122,11 +122,19 @@ input,
 $http-methods: ('get', lighten($blue, 5%)), ('put', lighten($orange, 5%)),
   ('post', lighten($green, 5%)), ('delete', lighten($red, 8%)),
   ('head', lighten($purple, 12%)), ('patch', $teal),
-  ('options', lighten($dark-blue, 15%));
+  ('options', lighten($dark-blue, 15%)),
+  ('propfind', lighten($custom-gray, 43%)), ('proppatch', lighten($custom-gray, 43%)),
+  ('copy', lighten($custom-gray, 43%)), ('move', lighten($custom-gray, 43%)),
+  ('lock', lighten($custom-gray, 43%)), ('unlock', lighten($custom-gray, 43%)),
+  ('mkcol', lighten($custom-gray, 43%));
```
Comment (daniel-schulz) [inline packages/desktop/src/renderer/styles/styles.scss:129]: @255kb I didn't found any usages of the variable `$http-methods` only for `$http-methods-light`.
But to be sure if I had overlooked a usage I've added the decent darker colors here (and also checked the contrast).

**Comment by daniel-schulz** [comment]
Do we need automated tests for this trivial change?
I didn't find any existing desktop test regarding HTTP methods.

File: packages/desktop/src/renderer/styles/styles.scss (hunk 1 above)
Comment (255kb) [reply packages/desktop/src/renderer/styles/styles.scss:129]: It's not used anymore/yet. I usually remove unused things, but I was lazy this time I guess. 
Thank you for updating it.

**Review by 255kb — APPROVED** [review]

**Comment by 255kb** [comment]
> Do we need automated tests for this trivial change? I didn't find any existing desktop tests for specific HTTP methods. Just some basic ones like `should be able to select a method by clicking` and `should be able to select a method by navigating with keyboard`.

I think it's ok. There is not much to test as these words are passed "as-is" to Node.js

Thank you for the contribution and for updating the contributing guidelines. 

Can you fix the linting/formatting? I will then merge and it will be included in the next release.

**Review by 255kb — CHANGES_REQUESTED** [review]
linting

File: packages/desktop/src/renderer/styles/styles.scss (hunk 1 above)
Comment (daniel-schulz) [reply packages/desktop/src/renderer/styles/styles.scss:129]: Ok, then I'll remove it together with the linting.

**Comment by daniel-schulz** [comment]
I've triggered the ci pipeline in my fork and fixed all the upcoming issues.
=> This PR should be merge ready - finally! :-)

**Comment by 255kb** [comment]
> Strange, the command `npm run lint` didn't gave any hints about the issues: image
> 
> Also the command `npm run format` didn't seem to work properly on my machine: image
> 
> But I manually executed prettier on the changed files and hope the CI linter will accept it now.
> 
> Can you re-trigger the CI pipeline?

Yeah sorry the formatting was failing not the linting. But the error you get is really strange, the command usually works (I use it in multiple env, Windows, Linux , etc.). I have no idea what happened.

Thanks for fixing everything 🙂

**Review by 255kb — APPROVED** [review]

**Pull request merged**